### ISSUE TITLE
[pickers] Disabled `DatePicker` contains two elements with `tabIndex=0`

### ISSUE THREAD

**Issue by erikastiebel** [mui/mui-x#19255]
### Steps to reproduce

Steps:
1. Open this link to live example: https://codesandbox.io/embed/m6gfm7
2. Open developer tools and add a live expression for document.activeElement
3. Click with the cursor just above the disabled date picker and press the tab-key several times
4. You can se from the live expression that there are two elements inside the disabled picker that captures focus, even if it's not visible to the eye.

### Current behavior

If you tab into a disablad datepicker, there are two elements that have tabindex=0.
First element is: div.MuiPickersSectionList-root.MuiPickersInputBase-sectionsContainer.css-1ugn09r-MuiPickersSectionList-root-MuiPickersInputBase-sectionsContainer-MuiPickersOutlinedInput-sectionsContainer)
and the second element is: span#(randomized id)-year.MuiPickersSectionList-sectionContent.MuiPickersInputBase-sectionContent.css-1a60ae4-MuiPickersSectionList-sectionContent-MuiPickersInputBase-sectionContent

### Expected behavior

If a datepicker is disabled I expect it to not be in the tab order.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: datepicker disabled tabindex accessibility

**Order ID**: 114708

**Comment by michelengelen** [comment]
Hey @erikastiebel ... it took a while, but I managed to find a solution for this:

```diff
--- a/packages/x-date-pickers/src/internals/hooks/useField/useFieldRootProps.ts
+++ b/packages/x-date-pickers/src/internals/hooks/useField/useFieldRootProps.ts
@@ -175,7 +175,7 @@ export function useFieldRootProps(

     // Other
     contentEditable: parsedSelectedSections === 'all',
-    tabIndex: parsedSelectedSections === 0 ? -1 : 0, // TODO: Try to set to undefined when there is a section selected.
+    tabIndex: internalPropsWithDefaults.disabled || parsedSelectedSections === 0 ? -1 : 0, // TODO: Try to set to undefined when there is a section selected.
   };
 }
```

```diff
--- a/packages/x-date-pickers/src/internals/hooks/useField/useFieldSectionContentProps.ts
+++ b/packages/x-date-pickers/src/internals/hooks/useField/useFieldSectionContentProps.ts
@@ -188,7 +188,7 @@ export function useFieldSectionContentProps(
         'aria-disabled': disabled,

         // Other
-        tabIndex: isContainerEditable || sectionIndex > 0 ? -1 : 0,
+        tabIndex: !isEditable || isContainerEditable || sectionIndex > 0 ? -1 : 0,
         contentEditable: !isContainerEditable && !disabled && !readOnly,
         role: 'spinbutton',
         id: `${id}-${section.type}`,
```

I will open this up as a good first issue.
Thanks for raising this! 🙇🏼
